### PR TITLE
uploading-assignment-4-conatains-question-based-on-array-and-string

### PR DESCRIPTION
**Pull request by munendra-successive**
1. Write a program to remove duplicate charecters from a string ("Hello Yellow" => "Helo Yw")
2.Write a program to reverse a string ("Hello John" => "olleH nhoJ")
3. Write a program to give count of variables in a string ("abcabcdabbcc" => "a3b4c4d1")
4. Write a program to convert given string to upperCase OR lowerCase
5. Write a program to convert given string in camel case ("hello john doe" => helloJohnDoe)
6. Write a program to find sum of an array
7. Write a program to find index of duplicate elements in an array
8. Write a program to remove dupliacte elements from an array
9. Write a program to find the last duplicate index in an array
10. Write a program to concatenate an input of array of arrays
11. Write a program to sort an array (bubble sort)
12. Write a program to display intersection of two array
13. Write a program to return inverse of an array
14. Write a program to reverse an array

### CODE REVIEW

File: assignment-4/Question-13.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Write a program to return inverse of an array
+
+
+// [4,2,3,0,1]
```
Comment (UrvashiChaudhary) [inline assignment-4/Question-13.js:4]: This is not working as expected
Expected: [1,0,3,2,4]
Actual output: [ '4', '2', '3', '0', '1' ]

File: assignment-4/Question-2.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+ return str;
+}
+
+let ans="";
+for(let index in str)
+{
+    ans+=reverseString(str[index]);
+}
+console.log(ans);
```
Comment (UrvashiChaudhary) [inline assignment-4/Question-2.js:25]: This is not working as expected, please look into it and try to solve it in a single loop

File: assignment-4/Question-3.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        {
+            if(str[index]==str[index2])
+            {
+                count++;
+            }
+        }
+        ans+=str[index]+count;
+    }
+}
+console.log(ans);
```
Comment (UrvashiChaudhary) [inline assignment-4/Question-3.js:21]: please try to reduce its complexity

File: assignment-4/Question-4.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+// Write a program to convert given string to upperCase OR lowerCase
+ 
+
+
+// into lowercase
+/*
+below code will convert a given string into lowercase by using ASCII codes of a character
```
Comment (UrvashiChaudhary) [inline assignment-4/Question-4.js:7]: try to solve it without ASCII codes

File: assignment-4/Question-5.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// Write a program to convert given string in camel case ("hello john doe" => helloJohnDoe)
+
+/*
```
Comment (UrvashiChaudhary) [inline assignment-4/Question-5.js:3]: Solve it without ASCII codes

File: assignment-4/Question-7.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    if(!temp.includes(arr[index]))
+    {
+        temp.push(arr[index]);
+    }
+    else
+    {
+        dupIndex.push(index);
+    }
+}
+console.log(dupIndex);
```
Comment (UrvashiChaudhary) [inline assignment-4/Question-7.js:17]: not working as expected

**Review by UrvashiChaudhary — COMMENTED** [review]
Please use proper indentation

File: assignment-4/Question-13.js (hunk 1 above)
Comment (munendra-successive) [reply assignment-4/Question-13.js]: Modified and giving output as expected

File: assignment-4/Question-2.js (hunk 2 above)
Comment (munendra-successive) [reply assignment-4/Question-2.js]: Modified the code and working fine with single iteration

File: assignment-4/Question-3.js (hunk 3 above)
Comment (munendra-successive) [reply assignment-4/Question-3.js]: Modified the code to reduced the complexity

File: assignment-4/Question-4.js (hunk 4 above)
Comment (munendra-successive) [reply assignment-4/Question-4.js]: Done without using ASCII codes

File: assignment-4/Question-5.js (hunk 5 above)
Comment (munendra-successive) [reply assignment-4/Question-5.js]: Done without using ASCII codes

File: assignment-4/Question-7.js (hunk 6 above)
Comment (munendra-successive) [reply assignment-4/Question-7.js]: Modified and giving expected output

**Pull request merged**